test/convert/replacements.test.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import { expect } from 'chai';
+import Sinon = require('sinon');
+import { Lifecycle } from '@salesforce/core';
+import { getReplacements, matchesFile, replacementIterations } from '../../src/convert/replacements';
+import { matchingContentFile } from '../mock';
+import * as replacementsForMock from '../../src/convert/replacements';
+
+describe('file matching', () => {
```

Comment:
Given that regexes are loaded from json, there will be nothing that verifies regexes while modifying the project json, allowing an invalid regex to be saved. At runtime the, the regex will fail parse. How will this error bubble out with enough information to properly identify the source of the error?

Reply:
```
ERROR running force:source:deploy:  Component conversion failed: Invalid regular expression: /BOOM\/: \ at end of pattern
```

How's that?